crates/storage/src/schema/revision_0057.rs

```diff
@@ -0,0 +1,1637 @@
+            block_number INTEGER NOT NULL
+        );
+        CREATE INDEX transaction_hashes_block_number ON transaction_hashes(block_number);
+        ",
+    )
+    .context("Creating new tables and indices")?;
+    let mut query_stmt = tx.prepare_cached(
+        r"
+        SELECT hash, tx, block_number, tx, receipt, events FROM starknet_transactions ORDER BY block_number, idx
+        ",
+    )?;
+    let mut insert_transaction_stmt = tx.prepare_cached(
+        r"INSERT INTO transactions (block_number, transactions, events) VALUES (?, ?, ?)",
+    )?;
+    let mut insert_transaction_hash_stmt =
+        tx.prepare_cached(r"INSERT INTO transaction_hashes (hash, block_number) VALUES (?, ?)")?;
+    const BATCH_SIZE: u32 = 10_000;
+
+    const LOG_RATE: Duration = Duration::from_secs(10);
+    let mut last_log = std::time::Instant::now();
+
+    let mut rows = query_stmt.query([])?;
+    let mut progress = 0;
+    let mut current_block = 0i64;
+    let mut transactions_in_block = Vec::new();
+    let mut events_in_block = Vec::new();
+    loop {
+        let mut batch_size = 0;
+        loop {
+            match rows.next() {
+                Ok(Some(row)) => {
+                    let hash = row.get_ref_unwrap("hash").as_blob()?;
+                    let block_number = row.get_ref_unwrap("block_number").as_i64()?;
+                    let transaction = row.get_ref_unwrap("tx").as_blob()?.to_vec();
+                    let receipt = row.get_ref_unwrap("receipt").as_blob()?.to_vec();
+                    let events = row.get_ref_unwrap("events").as_blob()?.to_vec();
+                    insert_transaction_hash_stmt.execute(params![hash, block_number])?;
+                    if block_number != current_block {
+                        transform_tx
+                            .send((
+                                current_block,
+                                mem::take(&mut transactions_in_block),
+                                mem::take(&mut events_in_block),
+                            ))
+                            .context("Sending transactions to transformer")?;
+                        current_block = block_number;
+                        batch_size += 1;
+                        if batch_size == BATCH_SIZE {
+                            break;
+                        }
+                    }
+                    transactions_in_block.push((transaction, receipt));
+                    events_in_block.push(events);
+                }
+                Ok(None) => {
+                    if !transactions_in_block.is_empty() {
+                        transform_tx
+                            .send((
+                                current_block,
+                                mem::take(&mut transactions_in_block),
+                                mem::take(&mut events_in_block),
+                            ))
+                            .context("Sending transactions to transformer")?;
+                    }
```

Comment:
We're not actually reading the result though if we don't increase `batch_size`.

```suggestion
                        batch_size += 1;
                    }
```